workers/scanner.go

```diff
@@ -399,6 +425,36 @@ func (s *Scanner) managedRefreshMinPinners() {
 	s.mu.Unlock()
 }
 
+// staticEligibleToPin returns true when this server is either pinning less than
+// AlwaysPinThreshold data or it's in the lower PinningRangeThresholdPercent
+// section of servers when ordered by contract data in descending order.
+// A server is always eligible if it's last in the list.
+func (s *Scanner) staticEligibleToPin(ctx context.Context) (bool, error) {
+	pinnedData, err := s.staticDB.ServerLoad(ctx, s.staticServerName)
+	if err != nil {
+		return false, err
+	}
+	// Below the hard limit.
+	if pinnedData < int64(AlwaysPinThreshold) {
+		return true, nil
+	}
+	pos, total, err := s.staticDB.ServerLoadPosition(ctx, s.staticServerName)
+	if err != nil {
+		return false, err
+	}
+	// Last in the list.
+	if pos == total {
+		return true, nil
+	}
+	// In the bottom PinningRangeThresholdPercent.
+	posPercent := 1.0 - float64(pos)/float64(total)
+	thresholdPercent := float64(PinningRangeThresholdPercent) / 100.0
+	if posPercent < thresholdPercent {
+		return true, nil
+	}
+	return false, nil
```

Comment:
In case you feel fancy `return posPercent < thresholdPercent, nil`

Reply:
Good point.